firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1789,14 +1789,19 @@ class BrowserViewController: UIViewController,
     private func executeToolbarActions() {
         guard isToolbarRefactorEnabled, let state = browserViewControllerState else { return }
 
-        if state.navigateTo != nil {
+        switch state {
```

Comment:
I chose to replace the series of if statements with a switch statement for a few reasons:
- `switch` statements can be easier to read than a series of if statements, especially when there are multiple conditions. Each case in a `switch` statement stands on its own, which can make the code more organized and easier to understand.
- If we need to add more conditions in the future and we will certainly need to, it's easier to add new cases to a `switch` statement than to add more `if` statements. This makes the code more maintainable and scalable.